packages/bappo-components/src/primitives/Switch/Switch.native/Switch.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import {
+  Animated,
+  NativeSyntheticEvent,
+  StyleSheet,
+  TargetedEvent,
+  TouchableOpacity,
+} from 'react-native';
+import styled from 'styled-components';
+
+import { InputHandle } from '../../../input-handle';
+import { SwitchProps } from '../types';
+
+const Switch = React.forwardRef(
+  (
+    {
+      accessibilityLabel,
+      disabled = false,
+      onBlur,
+      onFocus,
+      onValueChange,
+      style,
+      testID,
+      value = false,
+    }: SwitchProps,
+    ref: React.Ref<InputHandle>,
+  ) => {
+    const containerRef = React.useRef<TouchableOpacity>(null);
+
+    React.useImperativeHandle(ref, () => ({
+      focus: () => {
+        if (containerRef && containerRef.current) containerRef.current.focus();
+      },
+      blur: () => {
+        if (containerRef && containerRef.current) containerRef.current.blur();
+      },
+    }));
+
+    const toggleHandleLeft = React.useRef(new Animated.Value(2)).current;
+
+    const [prevValue, setPrevValue] = React.useState(value);
+
+    React.useEffect(() => {
+      if (!!value !== !!prevValue) {
+        Animated.timing(toggleHandleLeft, {
+          toValue: value ? 26 : 2,
+          duration: 200,
+        }).start();
+
+        setPrevValue(value);
+      }
+    }, [value, prevValue]);
```

Comment:
Hi Emil, there's an eslint error here saying that `toggleHandleLeft` is not included in the dependency. I suggest that instead of reading `toggleHandleLeft.current` and assign it to a variable outside `useEffect`, you read it inside `useEffect`. The difference is if you assign it to a variable, `useEffect` will only have access to that value even if it changes later whereas if you read `toggleHandleLeft.current` inside `useEffect` you always have the latest value which is how the old code behaves (reading from `this.state` always gives you the latest value).

Reply:
Hi William, thanks a lot for the suggestion and explanation. I've made the suggested change to rather read the value of `toggleHandleLeft` in the useEffect hook. I also realised that eslint was not set up correctly on my machine.